src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1947,6 +1947,25 @@ def unregister_provider(cmd, resource_provider_namespace, wait=False):
     _update_provider(cmd.cli_ctx, resource_provider_namespace, registering=False, wait=wait)
 
 
+def register_management_group(cmd, resource_provider_namespace, management_group_id, wait=False):
+    import time
+    rcf = _resource_client_factory(cmd.cli_ctx)
+    r = rcf.providers.register_at_management_group_scope(resource_provider_namespace, management_group_id)
```

Comment:
I could see register mg is a separate function in SDK. But we still could consider add integrate mg id registration with `az provider register` command by adding one additional parameter `--mg`.
In this way, we could also utilize previous check code _update_provider because the only difference is operation name.

Reply:
I changed the logic so that mg register utilizes the _update_provider function. Do you think that's enough or would you still like to see MG register integrated into `az provider register`?